pkg/app/app.go

```diff
@@ -159,3 +163,39 @@ func RealMain(opts types.Options, d types.Deployer, tester types.Tester) (result
 	}
 	return nil
 }
+
+func writeVersionToMetadataJSON(opts types.Options, d types.Deployer) error {
+	// setup the json metadata writer
+	metadataJSON, err := os.Create(
+		filepath.Join(opts.RunDir(), "metadata.json"),
+	)
+	if err != nil {
+		return err
+	}
+
+	meta, err2 := metadata.NewCustomJSON(nil)
+	if err2 != nil {
+		return err2
+	}
+	if err := meta.Add("kubetest-version", os.Getenv("KUBETEST2_VERSION")); err != nil {
```

Comment:
this currently tries to match with
https://github.com/kubernetes/test-infra/issues/20886#issuecomment-894586227